src/bundler.ts

```diff
@@ -408,23 +426,54 @@ class Bundler {
    *     output bundles. See 'polymer-analyzer/lib/bundle-manifest' for
    *     examples. UNUSED.
    */
-  async bundle(entrypoints: string[], strategy?: BundleStrategy):
-      Promise<DocumentCollection> {
-    console.assert(
-        entrypoints.length === 1,
-        'Only one entrypoint is supported, %d entrypoints were provided',
-        entrypoints.length);
-    const doc = await this._bundleDocument(
-        entrypoints[0], this.excludes, this.stripExcludes);
-    const collection = new Map<string, ASTNode>();
-    collection.set(entrypoints[0], doc);
-    return collection;
+  async bundle(
+      entrypoints: string[],
+      strategy?: BundleStrategy,
+      mapper?: BundleUrlMapper): Promise<DocumentCollection> {
+    const bundledDocuments = new Map<string, ASTNode>();
+    if (entrypoints.length === 1) {
+      const url = entrypoints[0];
+      debugger;
+      const depsIndex = await buildDepsIndex(entrypoints, this.analyzer);
+      debugger;
+      const bundles = generateBundles(depsIndex.entrypointToDeps);
+      for (const exclude of this.excludes) {
+        bundles[0].files.delete(exclude);
+      }
+      const manifest =
+          new BundleManifest(bundles, () => new Map([[url, bundles[0]]]));
+      const doc = await this._bundleDocument(url, bundles[0], manifest);
+      bundledDocuments.set(url, doc);
+      return bundledDocuments;
+    } else {
+      const bundles = new Map<string, ASTNode>();
+      if (!strategy) {
+        console.assert(strategy, 'strategy must be provided!');
+        throw new Error('strategy must be provided');
+      }
+      if (!mapper) {
+        console.assert(mapper, 'mapper must be provided!');
+        throw new Error(' mapper must be provided');
+      }
+      const index = await buildDepsIndex(entrypoints, this.analyzer);
+      const basicBundles = generateBundles(index.entrypointToDeps);
+      const bundlesAfterStrategy = strategy(basicBundles);
+      const manifest = new BundleManifest(bundlesAfterStrategy, mapper);
+      for (const bundleEntry of manifest.bundles) {
+        const bundleUrl = bundleEntry[0], bundle = bundleEntry[1];
+        const bundledAst = await this._bundleDocument(
+            bundleUrl, bundle, manifest, bundle.files);
+        bundledDocuments.set(bundleUrl, bundledAst);
+      }
+      return bundledDocuments;
+    }
   }
 
   private async _bundleDocument(
       url: string,
-      excludes: string[],
-      stripExcludes: string[]): Promise<ASTNode> {
+      bundle: Bundle,
+      bundleManifest: BundleManifest,
+      addedImports?: Set<string>): Promise<ASTNode> {
```

Comment:
This name leads me to think the imports provided in the set are additive (i.e. not represented in the document already)... is that the intent?  If so, [line 465](https://github.com/Polymer/vulcanize/pull/388/files#diff-841ba21ca2afd1b3c22749d1d7190395R465) where we hand `bundle.files` in as this argument was confusing to me.  I guess its called added because it might include some that aren't already in the doc.  But also, it defines whether inlining of an import could even happen, right?  So maybe its more like an `override` than an `added`?  Anyways, maybe add a comment?  Comment could just be above [line 509](https://github.com/Polymer/vulcanize/pull/388/files#diff-841ba21ca2afd1b3c22749d1d7190395R509) `if (addedImports) {`